source/opt/decoration_manager.cpp

```diff
@@ -305,12 +305,13 @@ void DecorationManager::RemoveInstructionFromTarget(ir::Instruction* inst,
                                                     const uint32_t target_id) {
   auto const group_iter = group_to_decoration_insts_.find(target_id);
   if (group_iter != group_to_decoration_insts_.end()) {
-    remove(group_iter->second.begin(), group_iter->second.end(), inst);
+    (void)std::remove(group_iter->second.begin(), group_iter->second.end(),
+                      inst);
```

Comment:
Consider
group_iter->second.erase(std::remove(group_iter->second.begin(), group_iter->second.end(), inst), group_iter->second.end());

Reply:
That physically resizes the array, which seems unnecessary.  I don't have strong intuition about whether that's significantly wasteful.  I'll make the change (logically, anyway).

Reply:
Vector, not array.